core/src/main/java/org/elasticsearch/search/suggest/term/TermSuggestionBuilder.java

```diff
@@ -65,12 +72,20 @@ public TermSuggestionBuilder(String name) {
      * tokens in the suggest text.
      * </ol>
      */
-    public TermSuggestionBuilder suggestMode(String suggestMode) {
+    public TermSuggestionBuilder suggestMode(@Nullable SuggestMode suggestMode) {
```

Comment:
I'm on the fence with this. In cases where the value is optional I don't think we need to make it a hard rule to check for null. We did it in several place in the queries, but there are also many setters without explicit null-checks, wdyt?